### ISSUE TITLE
selector-class-pattern firing warning on iterpolation

### ISSUE THREAD

**Issue by alan-agius4** [stylelint/stylelint#988]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

"selector-class-pattern is firing a waring on a class that is being generated through interpolation"

> Which rule, if any, is this issue related to?

`selector-class-pattern`

> What CSS is needed to reproduce this issue?

```css
@mixin alert-type-variant($type-variant) {
		.#{$type-variant} {
	           color: red;
	        }
	}
}
```

> What stylelint configuration is needed to reproduce this issue?
```json
        "selector-class-pattern": [
            "^([a-z]+[-]{0,2})+$",
            {
                resolveNestedSelectors: true
            }
        ]
```

> Which version of stylelint are you using?

`5.1.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

"PostCSS"

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

"No"

> What did you expect to happen?

"No warnings to be flagged."

> What actually happened (e.g. what warnings or errors you are getting)?

"The following warnings were flagged:"

```shell
test.css
 Expected class selector "." to match specified pattern (selector-class-pattern) [stylelint]
```

**Comment by jeddy3** [comment]
Thanks for the detailed report @alan-agius4 .

I think the solution here is to return early if the class selector contains interpolation. Would you like to contribute a PR with a fix? The same problem is solved in [`selector-no-id`](https://github.com/stylelint/stylelint/blob/master/src/rules/selector-no-id/index.js#L24) and [here are](https://github.com/stylelint/stylelint/blob/master/src/rules/selector-no-id/__tests__/index.js#L42-L81) the corresponding tests. It should just be a case of porting these over to `selector-class-pattern`. There's also a [small section](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-bugs) in the [working with rules developer guide](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md), which you might find helpful.

FYI, this is very much related to non-standard syntax btw :)

**Comment by alan-agius4** [comment]
Hi,

Yep, i can surly contribute to a PR later on today or tomorrow.
